src/models/documents.ts

```diff
@@ -55,8 +56,14 @@ export const DocumentsModel = types
             }
           });
 
-        return latestPublications
-          .sort((pub1, pub2) => parseInt(pub1.groupId!, 10) - parseInt(pub2.groupId!, 10));
+        return latestPublications.sort((pub1, pub2) => {
+          const user1 = clazz.getStudentById(pub1.uid);
+          const user2 = clazz.getStudentById(pub2.uid);
+          if (!user1 || !user2) return 0;
```

Comment:
I would think that documents without associated users should be sorted together, probably to the end, so something like:
```javascript
if (!user1 || !user2) return (user2 ? 1 : 0) - (user1 ? 1 : 0);
```

Reply:
Fixed, and added a test to double-check the logic. You had the right order!